cluster-autoscaler/cloudprovider/mcm/mcm_cloud_provider.go

```diff
@@ -423,7 +440,51 @@ func (machinedeployment *MachineDeployment) Nodes() ([]cloudprovider.Instance, e
 // Implementation optional.
 // TODO: add proper implementation
 func (machinedeployment *MachineDeployment) GetOptions(defaults config.NodeGroupAutoscalingOptions) (*config.NodeGroupAutoscalingOptions, error) {
-	return nil, cloudprovider.ErrNotImplemented
+	mcdSpec, err := machinedeployment.mcmManager.GetMachineDeploymentSpec(machinedeployment.Name)
+	if err != nil {
+		return nil, err
+	}
+	if mcdSpec == nil {
+		return nil, fmt.Errorf("nil machinedeployment returned for %s", machinedeployment.Name)
+	}
+
+	scaleDownUtilThresholdValue := defaults.ScaleDownUtilizationThreshold
+	if _, ok := mcdSpec.Annotations[ScaleDownUtilizationThresholdAnnotation]; ok {
+		if floatVal, err := strconv.ParseFloat(mcdSpec.Annotations[ScaleDownUtilizationThresholdAnnotation], 64); err == nil {
+			scaleDownUtilThresholdValue = floatVal
+		}
+	}
+	scaleDownGPUUtilThresholdValue := defaults.ScaleDownGpuUtilizationThreshold
+	if _, ok := mcdSpec.Annotations[ScaleDownGpuUtilizationThresholdAnnotation]; ok {
+		if floatVal, err := strconv.ParseFloat(mcdSpec.Annotations[ScaleDownGpuUtilizationThresholdAnnotation], 64); err == nil {
+			scaleDownGPUUtilThresholdValue = floatVal
+		}
+	}
+	scaleDownUnneededTimeValue := defaults.ScaleDownUnneededTime
+	if _, ok := mcdSpec.Annotations[ScaleDownUnneededTimeAnnotation]; ok {
+		if timeValue, err := time.ParseDuration(mcdSpec.Annotations[ScaleDownUnneededTimeAnnotation]); err == nil {
+			scaleDownUnneededTimeValue = timeValue
+		}
+	}
+	scaleDownUnreadyTimeValue := defaults.ScaleDownUnreadyTime
+	if _, ok := mcdSpec.Annotations[ScaleDownUnreadyTimeAnnotation]; ok {
+		if timeValue, err := time.ParseDuration(mcdSpec.Annotations[ScaleDownUnreadyTimeAnnotation]); err == nil {
+			scaleDownUnreadyTimeValue = timeValue
+		}
+	}
+	maxNodeProvisionTimeValue := defaults.MaxNodeProvisionTime
+	if _, ok := mcdSpec.Annotations[MaxNodeProvisionTimeAnnotation]; ok {
+		if timeValue, err := time.ParseDuration(mcdSpec.Annotations[MaxNodeProvisionTimeAnnotation]); err == nil {
+			maxNodeProvisionTimeValue = timeValue
+		}
+	}
+	return &config.NodeGroupAutoscalingOptions{
+		ScaleDownUtilizationThreshold:    scaleDownUtilThresholdValue,
+		ScaleDownGpuUtilizationThreshold: scaleDownGPUUtilThresholdValue,
+		ScaleDownUnneededTime:            scaleDownUnneededTimeValue,
```

Comment:
Can you rename `scaleDownUnneededTimeValue` to `scaleDownUnneededDuration`? Similar change for other values.